src/operators.jl

```diff
@@ -107,3 +113,69 @@ function Op(f, T=Any; iscommutative=false)
     finalizer(free, op)
     return op
 end
+
+"""
+    @RegisterOp(f, T, internal=false)
+
+Register a custom operator [`Op`](@ref) using the function `f` statically.
+On platfroms like AArch64, Julia does not support runtime closures,
+being passed to C. The generic version of [`Op`](@ref) uses that
+to support arbitrary function being passed as MPI reduction operators.
+In contrast `@RegisterOp` can be used to statically declare a function to
+be passed as an MPI operator.
+
+```julia
+function my_reduce(x, y)
+    2x+y-x
+end
+MPI.@RegisterOp(my_reduce, Int)
+# ...
+MPI.Reduce!(send_arr, recv_arr, my_reduce, MPI.COMM_WORLD; root=root)
+#...
+```
+!!! warning
+    Note that `@RegisterOp` works be introducing a new method to `Op`, potentially invalidating other users of `Op`.
```

Comment:
```suggestion
    Note that `@RegisterOp` works by introducing a new method to `Op`, potentially invalidating other users of `Op`.
```

Reply:
IMHO this is a little ambiguous. What does it mean to "invalidate other users of `Op`" in this context? Maybe an additional sentence or two when this might happen (or how to avoid it) would be helpful.